There is no definition of getdomainname() which for some reason work on Linux but does 
not compile on Solaris.

This is an upstream bug.
--- a/prnt/hpcups/HPCupsFilter.cpp	2019-04-03 18:32:40.233696275 +0200
+++ b/prnt/hpcups/HPCupsFilter.cpp	2019-04-03 18:34:29.308533471 +0200
@@ -42,6 +42,8 @@ #include <sys/utsname.h>
 #include <time.h>
 #include "utils.h"
 
+extern "C" int getdomainname(char*, int);
+
 #define HP_FILE_VERSION_STR    "03.09.08.0"
 
 static HPCupsFilter    filter;
